src/main/java/com/prgrms/vouchermanager/domain/PercentAmountVoucher.java

```diff
@@ -0,0 +1,36 @@
+package com.prgrms.vouchermanager.domain;
+
+import lombok.Getter;
+
+import java.util.UUID;
+
+@Getter
+public class PercentAmountVoucher implements Voucher {
+
+    private UUID id = UUID.randomUUID();
+    private final long percent;
+
+    public PercentAmountVoucher(long percent) {
+        this.percent = percent;
+    }
+
+    public PercentAmountVoucher(UUID id, long percent) {
+        this.id = id;
+        this.percent = percent;
+    }
+
```

Comment:
```suggestion
    private final UUID id;
    private final long percent;

    public PercentAmountVoucher(long percent) {
        this(UUID.randomUUID(), percent);
    }

    public PercentAmountVoucher(UUID id, long percent) {
        this.id = id;
        this.percent = percent;
    }
```

상태를 갖는 객체는 무조건 불변객체로 만들 수 있는 방법이 없을까에 대해서 고민하시는 것이 좋습니다.!
만약에 불변객체로 만들지 못한다면 그 후에 어떤식으로 필드 값 수정에 대해서 영향도를 줄일 수 있을까에 대해서 고민할 수 있을 것 같습니다.

불변객체가 아닌 경우, 해당 클래스가 어떠한 값을 갖고 있는지에 대해서 확신을 가질 수 없습니다.
어디서 정의되어 어디서 수정이 되었는지 등을 찾는 것보다 매번 불변 객체로 새롭게 만드는 것이 나을 것 같습니다.